src/gen9_avc_encoder.c

```diff
@@ -6648,6 +6648,51 @@ gen9_mfc_avc_insert_object(VADriverContextP ctx,
 }
 
 static void
+gen9_mfc_avc_insert_aud_packed_data(VADriverContextP ctx,
+                                    struct encode_state *encode_state,
+                                    struct intel_encoder_context *encoder_context,
+                                    struct intel_batchbuffer *batch)
+{
+    VAEncPackedHeaderParameterBuffer *param = NULL;
+    unsigned int length_in_bits;
+    unsigned int *header_data = NULL;
+    unsigned char *nal_type = NULL;
+    int count, i, start_index;
+
+    count = encode_state->slice_rawdata_count[0];
+    start_index = (encode_state->slice_rawdata_index[0] & SLICE_PACKED_DATA_INDEX_MASK);
+
+    for (i = 0; i < count; i++) {
+        unsigned int skip_emul_byte_cnt;
+
+        header_data = (unsigned int *)encode_state->packed_header_data_ext[start_index + i]->buffer;
+        nal_type = (unsigned char *)header_data;
+
+        param = (VAEncPackedHeaderParameterBuffer *)(encode_state->packed_header_params_ext[start_index + i]->buffer);
+
+        /* skip the slice header packed data type as it is lastly inserted */
+        if (param->type != VAEncPackedHeaderRawData && *(nal_type+4) != AVC_NAL_DELIMITER )
```

Comment:
Agreed with @yakuizhao , 
1. The start code prefix is 000001, not 00000001, so (nal_type+4) may not be the byte used for nal_unit_type.
2. It is legal to have some leading zero before the start code prefix, 000000000109
3. only the low 5bits is used as nal_unit_type